.circleci/config.yml

```diff
@@ -5,6 +5,9 @@ version: 2
     - image: ${IMAGE}
   steps:
     - checkout
+    - run:
+        name: Testing needed?
+        command: r=`git log --name-only --since="1 day ago" -n 2`; if [ "x$r" == "x" ]; then echo "No. Exiting."; exit -1; else echo "Yes."; fi
```

Comment:
-1 when no test needed is indeed suboptimal, since we'll got all kinds of build failures as a result.  

A little bit of Googling did not find any way of terminating the build without failures.  Only thing I can think of is to wrap every subsequent command in the same check as a way to skip them.

Reply:
Agree. The "successful" failures also annoyed me over the weekend. Either wrapping this code into a convenience script or shaming the CI provider into "CO2 friendliness" (there _should_ be a way to do nightly builds only if something truly changed). What does the OSX job do when no OSX is configured in the account?

Reply:
> What does the OSX job do when no OSX is configured in the account?

Not sure I really understand the question. We haven't used Mac OS X jobs in CircleCI before.